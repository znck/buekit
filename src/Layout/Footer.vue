<script>
import { style, styleResolver } from '../utils'
import PropTypes from '@znck/prop-types'

export default {
  name: 'Footer',
  functional: true,
  props: {
    tag: PropTypes.string.value('footer')
  },
  render (h, ctx = {}) {
    const _ = styleResolver(ctx.$style)
    const { tag } = ctx.props

    return style(
      h(tag, ctx.data, ctx.children),
      _('footer')
    )
  }
}
</script>

<style lang="scss" module>
@import '~bulma/sass/utilities/_all.sass';
@import '../_base.scss';
@import '~bulma/sass/layout/footer.sass';
</style>

