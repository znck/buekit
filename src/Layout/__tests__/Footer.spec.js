import {shallow} from 'vue-test-utils'
import {createRenderer} from 'vue-server-renderer'

import Footer from '../Footer.vue'

describe('Footer.vue', () => {
	it('matches snapshot', () => {
		const wrapper = shallow(Footer, {})
		const renderer = createRenderer()

		renderer.renderToString(wrapper.vm, (err, str) => {
			if (err) {
				throw new Error(err)
			}
			expect(str).toMatchSnapshot()
		})
	})
})
