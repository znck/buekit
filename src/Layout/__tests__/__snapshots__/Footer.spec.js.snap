// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer.vue matches snapshot 1`] = `<footer class="footer"></footer>`;
